python/GafferSceneUI/PathFilterUI.py

```diff
@@ -114,6 +118,56 @@
 
 )
 
+##########################################################################
+# VectorDataPlugValueWidget customisation
+###########################################################################
+
+class _PathsPlugValueWidget( GafferUI.VectorDataPlugValueWidget ) :
+
+	def __init__( self, plug, **kw ) :
+
+		GafferUI.VectorDataPlugValueWidget.__init__( self, plug, **kw )
+
+		self.vectorDataWidget().dataMenuSignal().connect( Gaffer.WeakMethod( self.__dataMenu ), scoped = False )
+
+	def __dataMenu( self, vectorDataWidget, menuDefinition ) :
+
+		selectedIndices = vectorDataWidget.selectedIndices()
+
+		if len( selectedIndices ) > 0 :
+			menuDefinition.append( "/divider", { "divider" : True } )
+			menuDefinition.append(
+				"/Select Filtered Objects",
+				{
+					"command" : functools.partial( self.__selectAffected, selectedIndices )
```

Comment:
I think we need a WeakMethod here. We own the VectorDataWidget, it owns the menu, and the menu points back to us via the command. If you open the menu, close the NodeEditor, and then change the value of `paths`, you currently get the dreaded `RuntimeError: Internal C++ object (_TableView) already deleted.`.

Reply:
Fixed in 3e96fded732a4d166ed6733e5b197d86d51ed308